### ISSUE TITLE
Questions about Yelp

### ISSUE THREAD

**Issue by hotchilipowder** [woni-seo/SiReN-reco#1]
Hello, thank you for offering this project for Sign-Aware Recommendation.

I have tried this code, it can achieve the results reported in the paper.
But I have some questions about the code and the baselines. Can you help me to fix it ?
1. The code uses [LightGCNConv](https://github.com/woni-seo/SiReN-reco/blob/591484d8950a8889e23a805578b2a6e0e59fa33f/convols.py#L10), but it does not do the [normalization](https://github.com/pyg-team/pytorch_geometric/blob/e97f8c23588e87ccfe2a3441d83e63fb243febfa/torch_geometric/nn/conv/gcn_conv.py#L184)?
2. I have some trouble for the baselines(e.g., LightGCN). I use [Recbole](https://recbole.io/docs/user_guide/config_settings.html) to run the split you offered in Yelp. 
But I have better metrics results than you reported in the paper.
In fact, it is closed to the results reported in LightGCN.

```
precision@10 : 0.0434    precision@15 : 0.03828    precision@20 : 0.03475    recall@10 : 0.06671    recall@15 : 0.08644    recall@20 : 0.10281    ndcg@10 : 0.06767    ndcg@15 : 0.07425    ndcg@20 : 0.08006
```
I paste the configs here if you are interested to run it in RecBole:

```
seed: 2020
# model config
embedding_size: 64
#n_layers: 2
#reg_weight: 1e-06

# dataset config
field_separator: "," 
seq_separator: " " 
USER_ID_FIELD: userId 
ITEM_ID_FIELD: itemId 
RATING_FIELD: rating 
TIME_FIELD: timestamp 
NEG_PREFIX: neg_ 
LABEL_FIELD: label 
load_col:
  inter: [userId, itemId, rating]

# training settings
epochs: 1000 
train_batch_size: 256 
learner: adam 
learning_rate: 0.001 
training_neg_sample_num: 1 
eval_step: 20 
stopping_step: 10

# evalution settings
metrics: ["Precision","Recall","NDCG"]
topk: [10, 15, 20] 
valid_metric: Precision@10
eval_batch_size: 4096 
loss_decimal_place: 4
metric_decimal_place: 5


# gpu settings
gpu_id: 0


# others for my task
benchmark_filename: ['train-1', 'test-1', 'test-1'] 

data_path: 'datas-0/'

show_progress: True

eval_args:
  split:
```

**Comment by cw-seo** [comment]
Hello, thank you for your interest. 

I have checked your comments. Let me explain about the questions.

 Q1-1) If you are talking about AGGREGATE function, we aggregate the neighbor nodes by using 'mean' aggregator by setting " aggr='mean' ". 
 Q1-2)If you are talking about edge weight (i.e., ratings), we do not use the ratings directly. Instead, we set a parameter w_o to distinguish "like" and "dislike" interactions (refer to section III-B).

 Q2) Did you remove the user-item interactions with 1, 2, 3 ratings? We remove user-item interactions with 1, 2, and 3 ratings by assuming that the interactions with 4, 5 ratings are preferable (refer to section V-D).

**Comment by hotchilipowder** [comment]
Hello, thank you for the reply.

Maybe I need to explain my question more clear.

1. I mean LightGCN layer is close to [Simplifying GCN](https://github.com/pyg-team/pytorch_geometric/blob/e97f8c23588e87ccfe2a3441d83e63fb243febfa/torch_geometric/nn/conv/sg_conv.py#L11).
It needs to do ```gcn_norm``` and use ```aggr=add``` to achieve the normalization coefficients. Your implementation is close to [```GraphSAGE```](https://github.com/pyg-team/pytorch_geometric/blob/e97f8c23588e87ccfe2a3441d83e63fb243febfa/torch_geometric/nn/conv/sage_conv.py#L11) instead of ```GCN```. 

2. I have remove the user-item interactions with 1,2,3 ratings. (I use your split by following codes). 
Firstly, I make .dat to .csv files.
```make .dat to .csv

import os
import pandas as pd

flists = [i for i in os.listdir('.') if i.endswith('.dat')]

for fname in flists:
    if fname.find('yelp')==-1: continue
    print(fname)
    a, _, c = fname.rpartition('.')
    a, _, b = a.rpartition('_yelp')
    print(a, b)
    fname2 = f'yelp/yelp.{a}-{b}.inter'
    #df = pd.read_csv(fname, header=None)
    df = pd.read_csv(fname)
    df.columns = ['0', 'userId:token', 'itemId:token', 'rating:float',]
    df['rating:float'] = df['rating:float'].astype(float)
    df = df.drop(['0'], axis=1)
    print(df)
    df = df[df['rating:float'] > 3.5]
    df.to_csv(fname2, index=None)
```
I move these  ```csv``` files to ```datas-0/yelp``` folder. 
And use the following script to run it in RecBole.
```  bole_general_yelp_test.yaml
seed: 2020
# model config
embedding_size: 64
#n_layers: 2
#reg_weight: 1e-06

# dataset config
field_separator: "," 
seq_separator: " " 
USER_ID_FIELD: userId 
ITEM_ID_FIELD: itemId 
RATING_FIELD: rating 
TIME_FIELD: timestamp 
NEG_PREFIX: neg_ 
LABEL_FIELD: label 
load_col:
  inter: [userId, itemId]

# training settings
epochs: 1000 
train_batch_size: 256 
learner: adam 
learning_rate: 0.001 
training_neg_sample_num: 1 
eval_step: 20 
stopping_step: 10

# evalution settings
metrics: ["Precision","Recall","NDCG"]
topk: [10, 15, 20] 
valid_metric: Precision@10
eval_batch_size: 4096 
loss_decimal_place: 4
metric_decimal_place: 5


# gpu settings
gpu_id: 0


# others for my task
benchmark_filename: ['train-1', 'test-1', 'test-1'] 

data_path: 'datas-0/'

show_progress: True

eval_args:
  split:

```
and run it by ```python run_bole_demo.py --config_files bole_general_yelp_test.yaml --dataset yelp```
```run_bole_demo.py
import argparse

from recbole.quick_start import run_recbole


if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('--model', '-m', type=str, default='LightGCN', help='name of models')
    parser.add_argument('--dataset', '-d', type=str, default='yelp', help='name of datasets')
    parser.add_argument('--config_files', type=str, default=None, help='config files')

    args, _ = parser.parse_known_args()

    config_file_list = args.config_files.strip().split(' ') if args.config_files else None
    run_recbole(model=args.model, dataset=args.dataset, config_file_list=config_file_list)

```
The result is ```{'precision@10': 0.04461, 'precision@15': 0.03931, 'precision@20': 0.03565, 'recall@10': 0.06846, 'recall@15': 0.08869, 'recall@20': 0.10589, 'ndcg@10': 0.06927, 'ndcg@15': 0.07601, 'ndcg@20': 0.08204}```.
